### PR TITLE
Add Java solution to Bronze Bovine Genomics (Solution 1)

### PR DESCRIPTION
**Pull request by Navigator365**
Also added to the Solution 1 description

### CODE REVIEW

File: solutions/usaco-736.mdx [hunk 1]
```diff
@@ -73,15 +73,78 @@ public class cownomics {
...
+		// Iterating through every character
+		for (int i = 0; i < M; i++) {
+			boolean duplicate = false;
+			// Iterating through every genome
+			for (int j = 0; j < N; j++) {
+				// Comparing to every other genome
+				for (int k = 0; k < N; k++) {
+					// If there are any duplicates,
+					// then this isn't a possible position,
+					// so we can stop counting and make sure we don't count it
```
Comment (SansPapyrus683) [inline solutions/usaco-736.mdx:125]:
```java
/* 
 * style multiline
 * comments like this
 */

File: solutions/usaco-736.mdx [hunk 2]
```diff
@@ -73,15 +73,78 @@ public class cownomics {
...
+					}
+				}
+			}
+			// If we haven't found any equal characters,
+			// then there are no duplicates and this is a possible sequence
+			if (!duplicate) {
+				possiblePositions++;
+			}
+		}
+
```
Comment (SansPapyrus683) [inline solutions/usaco-736.mdx:138]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
this code is the cleanest i've seen in weeks

**Review by SansPapyrus683 — APPROVED** [review]
epic

**Comment by Navigator365** [comment]
> epic
such an honor

**Comment by nathangong** [comment]
Since your solution is the expected solution for bronze, I think you could prob move it to Solution 1. The current Solution 1 is leaning closer to silver-level.

File: solutions/usaco-736.mdx [hunk 3]
```diff
@@ -73,15 +73,81 @@ public class cownomics {
 
 </LanguageSection>
 
-## Solution 2
+## Solution 2 - Brute force
 
-Brute force through every position and determine whether there are any characters that appear both in a spotty cow and a plain cow at this position.
+We'll iterate through every character in every genome, and determine whether, for each position, there are any characters that appear in any genome of a spotted and plain cow. If there are, this position isn't a possible solution, so we can end this part of our search and move on to the next position to save on runtime. If there aren't, this is a possible solution, so add one to our solution storage variable.
```
Comment (nathangong) [inline solutions/usaco-736.mdx:78]: I think "and determine whether, for each position, there are any characters that appear in any genome of a spotted and plain cow" is a bit confusing.

File: solutions/usaco-736.mdx [hunk 4]
```diff
@@ -122,3 +122,71 @@ file_out.close()
 </PySection>
 
 </LanguageSection>
+
+## Solution 2
```
Comment (nathangong) [inline solutions/usaco-736.mdx:126]:
```suggestion
## Solution 2 - HashSet
```

File: solutions/usaco-736.mdx [hunk 5]
```diff
@@ -122,3 +122,71 @@ file_out.close()
...
+
+## Implementation
+
+**Time Complexity**: $\mathcal O(N^2)$
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class cownomics {
```
Comment (nathangong) [inline solutions/usaco-736.mdx]:
```suggestion
public class BovineGenomics {
```

**Review by nathangong — APPROVED** [review]

File: solutions/usaco-736.mdx [hunk 6]
```diff
@@ -122,3 +122,69 @@ file_out.close()
...
+
+## Solution 2 - Hashset
+
+<LanguageSection>
+
+<JavaSection>
+
+## Implementation
+
+**Time Complexity**: $\mathcal O(N^2)$
```
Comment (ryanchou-dev) [inline solutions/usaco-736.mdx:134]:
```suggestion
**Time Complexity**: $\mathcal{O}(N^2)$
```

File: solutions/usaco-736.mdx [hunk 7]
```diff
@@ -2,87 +2,87 @@
...
-## Solution 1
+## Solution 1 - Brute force
 
-<LanguageSection>
-
-<JavaSection>
+We'll iterate through every character in every genome. For each character index, we'll find if there are any characters that appear in the genome of both a spotted and plain cow at that index. If there are, this position isn't a possible solution, so we can end this part of our search and move on to the next position to save on runtime. If there aren't, this is a possible solution. 
 
 ## Implementation
+**Time Complexity**: $\mathcal O(MN^2)
```
Comment (ryanchou-dev) [inline solutions/usaco-736.mdx:15]:
```suggestion
**Time Complexity**: $\mathcal{O}(MN^2)
```

**Review by ryanchou-dev — APPROVED** [review]
agree with sans, this code is a beauty ❤️. just fix the LaTeX :)

File: solutions/usaco-736.mdx (hunk 7 above)
Comment (nathangong) [reply solutions/usaco-736.mdx:15]: I edited the comment to include the colon inside of the bold style, as per https://usaco.guide/general/adding-solution#steps

File: solutions/usaco-736.mdx (hunk 6 above)
Comment (nathangong) [reply solutions/usaco-736.mdx:134]: Same here

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**